File: 백준/Silver/13305. 주유소/주유소.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./example.txt";
let input = fs.readFileSync(filePath).toString().split("\n");

let N = parseInt(input[0]);
let len = input[1].split(" ").map(BigInt);
let costs = input[2].split(" ").map(BigInt);
let sum = BigInt(0);
let min = BigInt(costs[0]);
let current_len = BigInt(0);

for (let i = 0; i < costs.length; i++) {
  let cost = costs[i];
    if (cost < min || i == N - 1) {
    sum = sum + current_len * min;
    min = cost;
    current_len = BigInt(0);
  }
  if (i != N - 1) current_len = current_len + BigInt(len[i]);
}
console.log(sum.toString());

File: 백준/Silver/13305. 주유소/주유소.py
n = int(input())

distance = list(map(int, input().split()))
price = list(map(int, input().split()))

minPrice = price[0]
ans = 0

for i in range(0, n-1):
    minPrice = min(minPrice, price[i])
    ans += (minPrice * distance[i])
print(ans)